README.md

```diff
@@ -0,0 +1,42 @@
+# cross_modality_query
+
+Server-side code for HuBMAP Cells API.
+[Python](https://github.com/hubmapconsortium/hubmap-api-py-client) and 
+[Javascript](https://github.com/hubmapconsortium/hubmap-api-js-client) clients available.
+
+## Usage
+
+This is only a high-level outline of the API:
+See the [Python client examples](https://github.com/hubmapconsortium/hubmap-api-py-client/tree/main/examples) for details.
+
+The API supports a small number of basic operations which can be combined to construct more complex queries.
+The API runs at a **`base_url`**: Currently `https://cells.dev.hubmapconsortium.org/api` is available.
+
+Issuing a `POST` to `{base_url}/{output_type}/` (where `output_type` is `cell`, `organ`, `gene`, or `cluster`),
+with query parameters in the body, will return a **query set handle**.
+(The Python and Javascript interfaces provide a **query set** abstraction, so you don't need to deal directly with the handle.)
+
+Issuing a `POST` to `{base_url}/{operation}/` (where `operation` is `union`, `intersection`, or `difference`),
+with query set handles provided as `key_one` and `key_two` in the body, will return a new query set handle,
+representing the result of the operation.
+
+Three endpoints are provided for getting more information, given a query set handle:
+- `{base_url}/count/` will return the number of matching records.
+- `{base_url}/evaluation/` will quickly return a limited set of fields from the records.
+- `{base_url}/detailevaluation/` can sort the results, and may return more fields, but may be a slower operation.
+
+To page through the results, `offset` and `limit` can be provided to `evaluation` and `detailevaluation`.
+
+## Coverage
+
+At this point, only some of the possible combinations of constraint type and output type have been implemented.
```

Comment:
Will it be possible to retrieve entities by filtering on properties of the entities themselves, such as their id or other properties (cell area, cluster name, etc.)?

Reply:
Queries by the entities id are in the works.  Queries by other properties of the entities themselves hasn't really been on my radar to this point.